### PR TITLE
feat: Add oauth2 flow to issuer

### LINKED ISSUE #4: Add oauth2 flow to issuer
Add oauth2 flow to issuer

1. Server will read oauth2 configuration from command line or environment variables
2. Add login handler
3. Add callback handler that will exchange code for token

For now return token.

### PR DESCRIPTION
**Pull request by sandrask**
-- server will read oauth2 configuration from command line or environment variables
-- add login handler
-- add callback handler that will exchange code for token

Closes #4

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: pkg/restapi/issuer/operation/operations.go [hunk 1]
```diff
@@ -27,30 +40,91 @@ type Handler interface {
...
+	if err != nil {
+		log.Error(err)
+		http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+
+		return
+	}
+
+	// validate that oauth cookie state matches the the state query parameter on your redirect callback
+	if r.FormValue(stateFormKey) != oauthState.Value {
+		log.Warn("invalid oauth state")
```
Comment (troyronda) [inline pkg/restapi/issuer/operation/operations.go]: This probably doesn't provide the event context.

File: pkg/restapi/issuer/operation/operations.go [hunk 2]
```diff
@@ -7,17 +7,30 @@ SPDX-License-Identifier: Apache-2.0
...
 const (
-	token = "/token"
+	login    = "/login"
+	callback = "/callback"
+
+	oauthCookieName = "oauthstate"
+	stateFormKey    = "state"
+	codeFormKey     = "code"
+
+	defaultCookieExpiry = 20
```
Comment (troyronda) [inline pkg/restapi/issuer/operation/operations.go]: comments on the unit type would be helpful. Is this minutes?

File: pkg/restapi/issuer/operation/operations.go [hunk 3]
```diff
@@ -27,30 +40,91 @@ type Handler interface {
...
-	svc := &Operation{}
+// Operation defines handlers for authorization service
+type Operation struct {
+	handlers  []Handler
+	cmsConfig *oauth2.Config
+}
+
+// Config defines configuration for issuer operations
+type Config struct {
+	CmsConfig *oauth2.Config
```
Comment (troyronda) [inline pkg/restapi/issuer/operation/operations.go]: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: pkg/restapi/issuer/operation/operations.go (hunk 1 above)
Comment (sandrask) [reply pkg/restapi/issuer/operation/operations.go]: No

**Pull request merged**